### PR TITLE
The redesign of comment details content

### PR DESCRIPTION
**Pull request by jarvislin**
See https://github.com/orgs/Automattic/projects/944?pane=issue&itemId=60382108
Design: yWt5gg3nWORhu079Qfv3mS-fi-1135_4600
Slack discussion: p1715785205781899/1715757510.947859-slack-C06BWNSR02K

This PR mainly implements the UI redesign of comment details content. I also removed lot of logic related to unused UI components at the moment, some of the logic will come back in modern Kotlin way when I handle the comment moderation tasks.

| Screenshot 1 | Screenshot 2 |
|---|---|
| Screenshot_20240516-151102 | Screenshot_20240516-151120 |

-----

## To Test:

1. Sign in JP app
2. Go to `Notifications` tab
3. Click on a notification with `Comment` type
4. The UI should be the same as the design on Figma
5. Click on the icon with three dot `...`
6. It should open the comment menu.
7. Click on `User Info`
8. It should display the user info bottom sheet.
9. The functionalities on the bottom sheet should be working as before.
10. Open the comment menu again.
11. Click on `Edit comment`.
12. It should navigate you to the screen of `Edit comment`.
13. The functionalities in this screen should be working as before.
14. Go to My Site -> More -> Comments, click on a comment
15. Repeat step 5 - 13.
16. Done, thank you!

P.S. The `User Info` bottom sheet displays different UI when we are from a comment or a notification because a notification contains richer information in its data.

-----

## Regression Notes

1. Potential unintended areas of impact

    - comment, notification

18. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

19. What automated tests I added (or what prevented me from doing so)

    - only updated some broken unit tests

-----

## PR Submission Checklist:

Skipped

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [x] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>strings.xml</code> files should only be updated on release branches, when the translations are downloaded by our automation.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/engagement/UserProfileBottomSheetFragment.kt
Comment (jarvislin) [inline WordPress/src/main/java/org/wordpress/android/ui/engagement/UserProfileBottomSheetFragment.kt:143]: I only moved the code snippets to this function, the logic is not changed.

File: WordPress/src/main/java/org/wordpress/android/ui/notifications/blocks/CommentUserNoteBlock.kt
Comment (jarvislin) [inline WordPress/src/main/java/org/wordpress/android/ui/notifications/blocks/CommentUserNoteBlock.kt:23]: Most of the changes are related to renaming variables and removing some unused functions in this file.

File: WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java
Comment (jarvislin) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java:217]: In this file, I removed a lot of logic related to `mActionBinding` which is no longer used.

File: WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt
Comment (justtwago) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt]: Super nit: here and in other places `.orEmpty()` function may look nicer (:
```suggestion
            userLogin = mComment?.authorEmail.orEmpty(),
```

File: WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt
Comment (justtwago) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt]: I'm not sure if it's safe to use the not-null assertion for `mNote`. Shouldn't we use the same approach as for the `mComment`?

File: WordPress/src/main/java/org/wordpress/android/ui/engagement/BottomSheetUiState.kt
Comment (justtwago) [inline WordPress/src/main/java/org/wordpress/android/ui/engagement/BottomSheetUiState.kt:7]: I'm wondering if we shouldn't mark some fields as nullable because not all are always available for all the cases. For example, if the user does not have an avatar URL, it should be `null` but not an empty string that has less straight meaning. When using the state in different places with not enough context, it looks like the state **has** to contain all the data (except for `onSiteClickListener`.)
Just leaving this note for your consideration :)

**Review by justtwago — APPROVED** [review]
Amazing work, Jarvis! 🚀 
I tested this on the Google Pixel 7 Pro, and it works perfectly. The only issue I found is that some images are not stretched as they are in the Reader, and custom emojis are sometimes not displayed. This is out of scope since it behaves the same way in trunk, but it’s worth mentioning anyway. 

Since the PR is pretty large, I would ask to wait for an extra thorough review from Antonis to make sure it works well and I haven’t missed anything.

File: WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt
Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt]: Yeah, it looks more elegant, I just updated.

File: WordPress/src/main/java/org/wordpress/android/ui/engagement/BottomSheetUiState.kt
Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/ui/engagement/BottomSheetUiState.kt:7]: I 100% agree with you, straight meaning is important for us to handle those fields correctly. I find many implementations use default values, but it'd be better if they are defined as nullable. 

I suspect that when handling this data with Java in the past, there was a tendency to use defensive programming to avoid crashes. However, now with Kotlin, handling nullables has become much easier. I think we can find time to refactor this.

File: WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt
Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt]: In `NotificationCommentDetailFragment`, `mNote` should be non-null after `onCreate()`.

I replaced with `note` here.

```
  private val note: Note // note will be non-null after onCreate
        get() = mNote!!
```

File: WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt:25]: Wdyt of using this getter for all the `mNote!!` occurrences in this file?

File: WordPress/src/main/java/org/wordpress/android/ui/comments/SiteCommentDetailFragment.kt
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/comments/SiteCommentDetailFragment.kt]: Nit pick: I wonder if we should avoid the non-null assertion operator for `mComment` in this file [since it is assigned a nullable value](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/comment-details-content/WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java#L412). Wdyt of using `?.` when possible with a default value?

**Review by antonis — APPROVED** [review]
Great work @jarvislin 👍 
I've tested the implementation on a Pixel 8 Pro (Android 14) and everything worked as described for me. The code also looks good and improves the codebase 🏅 
I've left a few nit picks related to non-null assertions but nothing blocking on my side. We can [revisit this separately as discussed](https://github.com/wordpress-mobile/WordPress-Android/pull/20798#discussion_r1606658173)

File: WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt
Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/ui/comments/NotificationCommentDetailFragment.kt:25]: Yeah, that makes sense, I missed some occurrences in this file when I made the change.

File: WordPress/src/main/java/org/wordpress/android/ui/comments/SiteCommentDetailFragment.kt
Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/ui/comments/SiteCommentDetailFragment.kt]: Hmmm, I just reviewed the `mComment`, I have a strong sense that `mComment` should be non-null in `SiteCommentDetailFragment`, but the parent class `CommentDetailFragment` handles `mComment` with defensive programming and make it like nullable.

I plan to replace it with

```
 private val comment: CommentModel
        get() = mComment!!
```

**Review by antonis — APPROVED** [review]
Thank you for the changes @jarvislin 🙇

**Comment by jarvislin** [comment]
Thank you both for reviewing this PR!

**Pull request merged**
